thefloatingdutchman/game_manager.py

```diff
@@ -47,6 +52,51 @@ def run(self, run_tutorial):
                 elif e.type == KEYDOWN and e.key == K_TAB:
                     # will eventually be moved
                     self._access_pause_screen()
+                elif e.type == KEYDOWN and e.key == K_e and self._room_manager.get_proximity():
+                    self._screen.fill("BLACK")
+
+                    s = Surface((1920, 1080))
+                    s.set_alpha(128)
+                    s.fill((0, 0, 0))
+                    self._screen.blit(s, (0, 0))
+                    upgradeChooser = random.choices([1, 2, 3], weights=[
+                        0.33, 0.33, 0.33], k=1)[0]
+                    if upgradeChooser == 1:
+                        old_att = self._player_manager.player._data._attack_speed
+                        self._player_manager._player._data._attack_speed += 1
+                        self._treasure_screen.update_treasure_screen(self._screen,"+1 to Attack Speed! Press any Key to Continue")
+
+                    elif upgradeChooser == 2:
+                        old_hel = self._player_manager.player._data._health
+
+                        self._player_manager.player._data._health += 1
+                        self._treasure_screen.update_treasure_screen(self._screen, "+1 to Health! Press any Key to Continue")
+
+                    elif upgradeChooser == 3:
+                        old_vel = self._player_manager.player._data._vel
+                        self._player_manager.player._data._vel += 1
+                        self._treasure_screen.update_treasure_screen(self._screen, "+1 to Velocity! Press any Key to Continue")
+
+                    display.flip()
+                    display.update()
+                    z =True
+                    while z:
+                        for e in event.get():
+                            if e.type == KEYDOWN:
+                                self._room_manager.set_cleared()
+
+                                upgradeChooser = random.choices([1, 2, 3], weights=[
+                                    0.33,0.33, 0.33], k=1)[0]
+                                if upgradeChooser == 1:
+                                    self._player_manager._player._data._attack_speed += 5
+                                if upgradeChooser == 2:
+                                    self._player_manager.player._data._health += 5
+                                if upgradeChooser == 3:
+                                    self._player_manager.player._data._vel += 5
+
+                                self._items_dropped = True
+                                z=False
+

```

Comment:
This seems like a lot to be shoving into game manager. It has its own UI while loop and drop logic which should be extracted to classes if possible